doc/specs/stdlib_array.md

```diff
@@ -10,8 +10,58 @@ title: array
 
 Module for index manipulation and array handling tasks.
 
-## Procedures and methods provided
+## Derived types provided
```

Comment:
Thank you for this PR. Could you open a PR specific for the additions related to `stdlib_array`, please? It will make the review of this PR easier. Let me know if you need help.